web/packages/quick-editor/src/quick-editor.ts

```diff
@@ -0,0 +1,43 @@
+import { GravatarQuickEditorCore, ProfileUpdatedType, Scope } from './quick-editor-core';
+
+export type QuickEditorOptions = {
+	email: string;
+	editorTriggerSelector: string;
+	avatarSelector: string;
+	scope?: Scope;
+	local?: string;
+}
+
+export default class GravatarQuickEditor {
+	_quickEditor: GravatarQuickEditorCore;
+	_editorTrigger: HTMLElement;
+	_avatarList: NodeListOf<HTMLImageElement>;
+
+	constructor( { email, editorTriggerSelector, avatarSelector, scope, local }: QuickEditorOptions ) {
+		this._quickEditor = new GravatarQuickEditorCore( { email, scope, local, onProfileUpdated: this._onProfileUpdated.bind( this ) } );
+		this._editorTrigger = document.querySelector( editorTriggerSelector );
+		this._avatarList = document.querySelectorAll( avatarSelector );
+
+		this._editorTrigger?.addEventListener('click', () => this._quickEditor.open() )
+	}
+
+	_onProfileUpdated( type: ProfileUpdatedType ) {
+		if ( type !== 'avatar_updated' || this._avatarList.length === 0 ) {
+			return;
+		}
+
+		this._avatarList.forEach( avatarElement => {
+			if ( ! URL.canParse( avatarElement.src ) ) {
+				return;
+			}
+
+			const avatarURL = new URL( avatarElement.src );
```

Comment:
The `URL.canParse( avatarElement.src )` right above seems enough as a safety check.

Reply:
Since we're helping users update their Gravatar's avatars. Do you think we should check if the `src` belongs to a Gravatar URL (i.e. `gravatar.com/avatar/`)?

Reply:
Checking the Gravatar URL would work in most of the scenarios.
But there's one specific scenario where it wouldn't work well:
Imagine a developer using their own default user profile image (not the Gravatar one) for users that don't have a defined avatar.
If they set this element to be updated after the user adds a profile image, the element wouldn't be updated because of the URL check.
It seems like an edge case, but it's not impossible to happen.

Reply:
> Imagine a developer using their own default user profile image (not the Gravatar one) for users that don't have a defined avatar.
> If they set this element to be updated after the user adds a profile image, the element wouldn't be updated because of the URL check.

If that's the case, the avatar's URLs are different (e.g. `https://abc.com/default-avatar` v.s. `https://secure.gravatar.com/avatar/123`), so developers seem to have to update the URL by themself. Am I understanding correctly?

Reply:
If we don't check the URL, developers wouldn't need to update the URL manually.
Honestly, I don't see a great benefit in checking the URL.

Reply:
> If we don't check the URL, developers wouldn't need to update the URL manually.

I might be wrong. As I understand:
 
- Developers use their default avatar image for an `img` element, e.g. `https://abc.com/default-avatar`
- The user updates their Gravatar's avatar. E.g. `https://secure.gravatar.com/avatar/123`
- Developers will need to change the URL from `https://abc.com/default-avatar` to `https://secure.gravatar.com/avatar/123`
- I don't think the `https://abc.com/default-avatar?t=xxxx` (through the `avatarSelector` option) could work for this case

> Honestly, I don't see a great benefit in checking the URL.

I suppose the `avatarSelector` is used for Gravatar's URLs. But if you think it's not worth checking the URLs. I'm fine without it.

Reply:
You're right, sorry!
In my mind, I was thinking we were receiving the avatar URL from the PostMessage, but we are not.
I'll add the check.
Sorry for the confusion.

Reply:
Addressed on https://github.com/Automattic/gravatar/pull/14/commits/c0575eb75f05e9245b641d1fab5ffeac3ca43855